src/main/java/org/asamk/signal/manager/Manager.java

```diff
@@ -204,6 +205,14 @@ public void updateAccountAttributes() throws IOException {
         accountManager.setAccountAttributes(account.getSignalingKey(), account.getSignalProtocolStore().getLocalRegistrationId(), true, account.getRegistrationLockPin(), getSelfUnidentifiedAccessKey(), false);
     }
 
+    public void setProfileName(String name) throws IOException {
+        accountManager.setProfileName(account.getProfileKey(), name);
+    }
+
+    public void setProfileAvatar(File avatar) throws IOException {
+        accountManager.setProfileAvatar(account.getProfileKey(), Utils.createStreamDetailsFromFile(avatar));
```

Comment:
If I read the libsignal-service-java code correctly it doesn't close the input stream that gets passed here.
That means we need to store the streamDetails in a variable and call `streamDetails.getStream().close()` at the end of the method.